### PR TITLE
Parallel cloudfile diffs and converges

### LINKED ISSUE #203: Support converging multiple clouds at the same time
# Description
Users should be able to pass multiple Cloudfiles into commands such as `diff` and `converge`.

### PR DESCRIPTION
**Pull request by bcall-r7**
# Description
Fix #203 - This adds the ability to converge or diff multiple cloudfiles at the same time.

# Motivation and Context
Aids in deploying to multiple regions.
# Usage Examples
```
bundle exec convection diff --cloudfiles us-east-1/Cloudfile eu-central-1/Cloudfile
bundle exec convection converge --cloudfiles us-east-1/Cloudfile eu-central-1/Cloudfile
```
# Testing Steps
- [x] Diffs work with one cloudfile
- [x] Diffs work with two cloudfiles
- [x] Converges work with one cloudfile
- [x] Converges work with two cloudfiles
- [x] Converge/diff will check the current working directory for a cloudfile
- [x] Delayed output "true" will print output upon task completion
- [x] Delayed output "false" will print output immediately 

# Remaining Tasks
- [x] Update docs with new converge/diff examples
- [x] Update docs around stack print changes

### CODE REVIEW

File: bin/convection [hunk 1]
```diff
@@ -1,35 +1,64 @@
...
 
     desc 'delete STACK', 'Delete stack(s) from your cloud'
     option :stack_group, :type => :string, :desc => 'The name of a stack group defined in your cloudfile to delete'
     option :stacks, :type => :array, :desc => 'A ordered space separated list of stacks to delete'
+    option :cloudfile, :type => :string, :default => 'Cloudfile'
     option :verbose, :type => :boolean, :aliases => '--v', :desc => 'Show stack progress', default: true
     option :'very-verbose', :type => :boolean, :aliases => '--vv', :desc => 'Show unchanged stacks', default: true
     def delete(stack = nil)
+      init_cloud
       @cloud.configure(File.absolute_path(options['cloudfile'], @cwd))
```
Comment (erran-r7) [inline bin/convection]: Couldn't this line be in `#init_cloud` as well?

File: bin/convection [hunk 2]
```diff
@@ -74,29 +132,50 @@ module Convection
...
+
+      def print_info(say, region: nil)
+        print "#{region} " if region
+        say_status(*say.to_thor)
+      end
+
+      def print_outputs(outputs)
+        outputs.each do |output|
+          puts '********'
+          puts output[:region]
```
Comment (erran-r7) [inline bin/convection]: I'd love to use print the cloud name vs./in addition to the region.

File: bin/convection [hunk 3]
```diff
@@ -74,29 +81,83 @@ module Convection
 
       private
 
-      def emit_events(event, *errors)
+      def task(task_name, stack)
```
Comment (erran-r7) [inline bin/convection]: This gets confusing when you look at stack tasks such as the ones that are being extended in #207.

How about `#operation` or something else?

File: bin/convection [hunk 4]
```diff
@@ -52,20 +55,24 @@ module Convection
...
+      task('diff', stack)
+      exit 1 if @errors
     end
 
-    desc 'print STACK', 'Print the rendered template for STACK'
-    def print(stack)
-      @cloud.configure(File.absolute_path(options['cloudfile'], @cwd))
+    desc 'print_template STACK', 'Print the rendered template for STACK'
+    option :cloudfile, :type => :string, :default => 'Cloudfile'
+    def print_template(stack)
```
Comment (erran-r7) [inline bin/convection]: hehe, run into `print` overriding `Kernel#print`? 😜 I like `print-template` more anyways. Let's use hyphen over underscore in the documentation. Thor makes underscores interchangeable for dashes in commands as well as in flags. 👍

File: bin/convection (hunk 3 above)
Comment (bcall-r7) [inline bin/convection]: sounds good

File: bin/convection (hunk 4 above)
Comment (bcall-r7) [inline bin/convection]: sounds good

File: bin/convection [hunk 5]
```diff
@@ -92,6 +103,38 @@ module Convection
 
       private
 
+      def operation(task_name, stack)
+        work_q = Queue.new
+        semaphore = Mutex.new
+        if options[:cloudfiles].length > 1 && options[:delayed_output]
+          puts 'For easier reading when using multiple cloudfiles output will be delayed until task completion.'
```
Comment (erran-r7) [inline bin/convection]: ❤️

File: bin/convection [hunk 6]
```diff
@@ -1,36 +1,41 @@
 #!/usr/bin/env ruby
 require 'thor'
 require_relative '../lib/convection/control/cloud'
-
+require 'thread'
+require 'colorize'
```
Comment (erran-r7) [inline bin/convection]: 🔍 where do we use colorise?

**Pull request merged**